app/routes/route_utilities.py

```diff
@@ -15,4 +15,29 @@ def validate_model(cls, model_id):
         response = {"message": f"{cls.__name__} {model_id} not found"}
         abort(make_response(response, 404))
     
-    return model
\ No newline at end of file
+    return model
+
+def create_model(cls, model_data):
+    try:
+        new_model = cls.from_dict(model_data)
+        
+    except KeyError as error:
+        response = {"message": f"Invalid request: missing {error.args[0]}"}
+        abort(make_response(response, 400))
+    
+    db.session.add(new_model)
+    db.session.commit()
+
+    return make_response(new_model.to_dict(), 201)
+
+def get_models_with_filters(cls, filters=None):
+    query = db.select(cls)
+    
+    if filters:
+        for attribute, value in filters.items():
+            query = query.where(getattr(cls, attribute).ilike(f"%{value}%"))
```

Comment:
Should we wrap this with a try/except block to skip attempts that raise AttributeError?

Reply:
That sounds good to make the function a little more sturdy.

Reply:
How does it sound to wrap it in a hasattr check so we don't try to apply a query if the column doesn't exist?
```py
if filters:
    for attribute, value in filters.items():
        if hasattr(cls, attribute):
            query = query.where(getattr(cls, attribute).ilike(f"%{value}%"))
```

Unless we want to abort the call if someone passes a misspelled or nonexistent column, I'm struggling with what we would want to do with the except side of a try/except.

Reply:
Yeah, it is not robust, but it's a reminder to at least consider error handling. We can frame it as a small but imperfect example of the kinds of abstractions we can use to DRY up our code, and if anyone wants to they should feel free to try breaking it and see what they could add to make it stronger.

Reply:
That sounds like a good framing to me.